main.tf

```diff
@@ -37,7 +37,7 @@ resource "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
 # Configure account public access
 # (Using restapi provider for this until official IBM provider support is added -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/3285)
 resource "restapi_object" "account_public_access" {
-  path           = "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
+  path           = var.private_endpoint ? "//private.iam.cloud.ibm.com/v2/groups/settings?account_id={id}" : "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
```

Comment:
can we move the code duplication here? Put `iam.cloud.ibm.com/v2/groups/settings?account_id={id}` into a local variable

Reply:
Done